lib/xmldoc.js

```diff
@@ -139,8 +143,12 @@ XmlElement.prototype.toStringWithIndent = function(indent, options) {
     if (Object.prototype.hasOwnProperty.call(this.attr, name))
         s += " " + name + '="' + escapeXML(this.attr[name]) + '"';
 
-  var finalVal = escapeXML(this.val.trim());
-
+  var finalVal = '';
+  if (this.isValCdata){
+    finalVal = '<![CDATA['+this.val+']]>';
+  } else {
+    finalVal = escapeXML(this.val);
```

Comment:
I forget if I commented on this before but this has lost the `trim()` from the previous implementation.